src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -989,7 +991,7 @@ public Response publishDataset(@PathParam("id") String id, @QueryParam("type") S
                  * 
                  */
                 if ((ds.getModificationTime()!=null && (ds.getIndexTime() == null || (ds.getIndexTime().compareTo(ds.getModificationTime()) <= 0))) ||
-                        (ds.getPermissionModificationTime()!=null && (ds.getIndexTime() == null || (ds.getIndexTime().compareTo(ds.getModificationTime()) <= 0)))) {
+                        (ds.getPermissionModificationTime()!=null && (ds.getIndexTime() == null || (ds.getIndexTime().compareTo(ds.getPermissionModificationTime()) <= 0)))) {
```

Comment:
Not sure, but you compared indexTime with modificationTime at the start, so this may be a copy-paste slip-up?